File: 00_Sorting_00/insertion_sort.c
/*
Name : Kumar Shanu
DAA LAB
//////////Program  Insertion Sort////////////////
*/

void insertion_sort(int arr[],int n){
    int i,j;

    for(i=1; i<n; i++){
        int key = arr[i];
        j= i-1;
        while(j>=0 && arr[j] > key){
            arr[j+1] = arr[j];
            j = j-1;
            arr[j+1] = key;
        }
    }
}

int main(void){
    int arr[20],n,item;
    printf("Enter the size of array: ");
    scanf("%d",&n);

    printf("Enter the elements of array: ");

    for(int i=0; i<n; i++)
        scanf("%d",&arr[i]);

    printf("\nThe array is : \n");
    for(int i=0; i<n; i++)
        printf("%4d",arr[i]);

    insertion_sort(arr,n);
    printf("\nAfter Sorting The array is : \n");
    for(int i=0; i<n; i++)
        printf("%4d",arr[i]);
}

File: Algorithms_py/insertion_sort.py
def insertion_sort(elements: list) -> None:
    """Sort the list using insertion sort algorithm.

    Args:
        elements (list): list of elements
    """
    for i in range(1, len(elements)):
        anchor = elements[i]
        j = i - 1
        while j >= 0 and anchor < elements[j]:
            elements[j + 1] = elements[j]
            j -= 1
        elements[j + 1] = anchor


if __name__ == "__main__":
    arr = [eval(x) for x in input("Enter the array: ").split()]
    print("Array is: ", arr)
    insertion_sort(arr)
    print("Array is: ", arr)
